core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2186,13 +2186,22 @@ class SparkContext(config: SparkConf) extends Logging {
    * has overridden the call site using `setCallSite()`, this will return the user's version.
    */
   private[spark] def getCallSite(): CallSite = {
-    lazy val callSite = Utils.getCallSite()
-    CallSite(
-      Option(getLocalProperty(CallSite.SHORT_FORM)).getOrElse(callSite.shortForm),
-      Option(getLocalProperty(CallSite.LONG_FORM)).getOrElse(callSite.longForm)
-    )
+    if (getLocalProperty(CallSite.SHORT_FORM) == null
```

Comment:
@lxian, it would be great if we can elabourate how it causes the lock as you did in JIRA. In addition, to clarify, this is not an issue in Spark 3+ as we dropped Scala 2.11 at SPARK-26132.

Can you create a PR for bracnh-2.4 alone? cc @viirya and @dongjoon-hyun FYI

Reply:
Is this easy to add test? Otherwise we can also show up how to reproduce it manually in the description.

Reply:
@lxian Can you create a PR for branch-2.4? If you are busy, would you mind I create a PR for branch-2.4? Thanks.

Reply:
@viirya I believe it's fine for you to just go ahead IMO .. the author became inactive 4 days and this is the blocker of 2.4 (I guess?).

Reply:
This is the last issue for 2.4, I think. Okay, let me create a PR first. I can close it if the author opens his after.

Reply:
@viirya Thank you for creating the PR. you can close mine if needed.

Reply:
Thank you @lxian. Your authorship is still kept in the commit.